exotica_core/src/tools/conversions.cpp

```diff
@@ -89,21 +90,19 @@ KDL::Frame GetFrameFromMatrix(Eigen::MatrixXdRefConst val)
 
 Eigen::MatrixXd GetFrame(const KDL::Frame& val)
 {
-    Eigen::MatrixXd ret = Eigen::MatrixXd::Identity(4, 4);
-    ret(0, 3) = val.p(0);
-    ret(1, 3) = val.p(1);
-    ret(2, 3) = val.p(2);
-    ret.block(0, 0, 3, 3) = Eigen::Map<const Eigen::Matrix3d>(val.M.data).transpose();
-    return ret;
+    Eigen::Affine3d ret;
```

Comment:
Please use Isometry, not Affine.

Reply:
Any reason for this?

Reply:
Yes. If you would do an inverse on the Affine, it runs SVD which is expensive. Rigid body motions do not do shearing, scaling etc. so no need to have. Isometry is much cheaper with the inverse as it's direct.